File: workshop/src/main/java/models/ReservationMusee.java
package models;

import java.util.Date;

public class ReservationMusee {
    private int reservation_id;
    private Date date_reservation;
    private int nbr_tickets_reserves;
    private User user;
    private Musee musee;

    public ReservationMusee() {}
    public ReservationMusee(int reservation_id,Date date_reservation,int nbr_tickets_reserves,User user,Musee musee){
        this.reservation_id = reservation_id;
        this.date_reservation = date_reservation;
        this.nbr_tickets_reserves = nbr_tickets_reserves;
        this.user = user;
        this.musee = musee;

    }
    public ReservationMusee(Date date_reservation,int nbr_tickets_reserves,User user,Musee musee){
        this.date_reservation = date_reservation;
        this.nbr_tickets_reserves = nbr_tickets_reserves;
        this.user = user;
        this.musee = musee;

    }
    public ReservationMusee(Date date_reservation,int nbr_tickets_reserves,Musee musee){
        this.date_reservation = date_reservation;
        this.nbr_tickets_reserves = nbr_tickets_reserves;
        this.musee = musee;

    }
    public String getUsername(){
        return user.getUser_name();
    }
    public String getNom_musee(){
        return musee.getNom_musee();
    }

    public int getReservation_id() {
        return reservation_id;
    }

    public void setReservation_id(int reservation_id) {
        this.reservation_id = reservation_id;
    }

    public Date getDate_reservation() {
        return date_reservation;
    }

    public void setDate_reservation(Date date_reservation) {
        this.date_reservation = date_reservation;
    }

    public int getNbr_tickets_reserves() {
        return nbr_tickets_reserves;
    }

    public void setNbr_tickets_reserves(int nbr_tickets_reserves) {
        this.nbr_tickets_reserves = nbr_tickets_reserves;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Musee getMusee() {
        return musee;
    }

    public void setMusee(Musee musee) {
        this.musee = musee;
    }
}

File: src/Entity/ReservationMusee.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * ReservationMusee
 *
 * @ORM\Table(name="reservation_musee", indexes={@ORM\Index(name="fk_musee_reservation", columns={"id_musee"}), @ORM\Index(name="fk_user_reservation", columns={"id_user"})})
 * @ORM\Entity
 */
class ReservationMusee
{
    /**
     * @var int
     *
     * @ORM\Column(name="reservation_id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $reservationId;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_reservation", type="date", nullable=false)
     */
    private $dateReservation;

    /**
     * @var int
     *
     * @ORM\Column(name="nbr_tickets_reserves", type="integer", nullable=false)
     */
    private $nbrTicketsReserves;

    /**
     * @var \Musee
     *
     * @ORM\ManyToOne(targetEntity="Musee")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_musee", referencedColumnName="id_musee")
     * })
     */
    private $idMusee;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_user", referencedColumnName="user_id")
     * })
     */
    private $idUser;

    public function getReservationId(): ?int
    {
        return $this->reservationId;
    }

    public function getDateReservation(): ?\DateTimeInterface
    {
        return $this->dateReservation;
    }

    public function setDateReservation(\DateTimeInterface $dateReservation): self
    {
        $this->dateReservation = $dateReservation;

        return $this;
    }

    public function getNbrTicketsReserves(): ?int
    {
        return $this->nbrTicketsReserves;
    }

    public function setNbrTicketsReserves(int $nbrTicketsReserves): self
    {
        $this->nbrTicketsReserves = $nbrTicketsReserves;

        return $this;
    }

    public function getIdMusee(): ?Musee
    {
        return $this->idMusee;
    }

    public function setIdMusee(?Musee $idMusee): self
    {
        $this->idMusee = $idMusee;

        return $this;
    }

    public function getIdUser(): ?User
    {
        return $this->idUser;
    }

    public function setIdUser(?User $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }


}
